XamlControlsGallery/ControlPages/ListViewPage.xaml.cs

```diff
@@ -56,6 +85,242 @@ private void SelectionModeComboBox_SelectionChanged(object sender, SelectionChan
                 }
             }
         }
+
+        //===================================================================================================================
+        // Drag/Drop Example
+        //===================================================================================================================
+
+        private void Source_DragItemsStarting(object sender, DragItemsStartingEventArgs e)
+        {
+            // Prepare a string with one dragged item per line
+            StringBuilder items = new StringBuilder();
+            foreach (Contact item in e.Items)
+            {
+                if (items.Length > 0) { items.AppendLine(); }
+                if (item.ToString() != null)
+                {
+                    // Append name from contact object onto data string
+                    items.Append(item.ToString() + " " + item.Company);
+                } 
+            }
+            // Set the content of the DataPackage
+            e.Data.SetText(items.ToString());
+
+            e.Data.RequestedOperation = DataPackageOperation.Move;
+
+        }
+
+        private void Target_DragOver(object sender, DragEventArgs e)
+        {
+            e.AcceptedOperation = DataPackageOperation.Move;
+        }
+
+        private void Source_DragOver(object sender, DragEventArgs e)
+        {
+            e.AcceptedOperation = DataPackageOperation.Move;
+        }
+
+        private async void Target_Drop(object sender, DragEventArgs e)
+        {
+            if (e.DataView.Contains(StandardDataFormats.Text))
+            {
+
+                DragOperationDeferral def = e.GetDeferral();
+                string s = await e.DataView.GetTextAsync();
+                string[] items = s.Split('\n');
```

Comment:
Interesting. Maybe @ranjeshj can comment on whether this is the best way to initiate (drag-and-)drop? I really know nothing about this area, but what's bothering me a little is the need to create a new temporary Contact out of strings. Is there any way to pass an object directly?

Reply:
I think my reasoning for doing this was something about not being able to send data packages containing something other than a string. Would prefer to pass the object directly if possible!

Reply:
Drag and drop is a strange and magical feature, with mysterious dragons lurking about. For more context, consider checking out the TabView drag and drop example.